handler.go

```diff
@@ -95,14 +97,25 @@ func (h *Handler) Provision(ctx caddy.Context) error {
 				finalReplace := placeholderRepl.ReplaceKnown(repl.Replace, "")
 
 				if repl.re != nil {
-					tr := replace.RegexpString(repl.re, finalReplace)
+					// This would require the following method in the upstream icholy/replace/replace.go
+					// func RegexpStringWithLateBinding(re *regexp.Regexp, template string, 
+					// 	binding func(string) string) *RegexpTransformer {
+					// 		return RegexpIndexFunc(re, func(src []byte, index []int) []byte {
+					// 			return re.Expand(nil, []byte(binding(template)), src, index)
+					// 	})
+					// }
+					tr := replace.RegexpStringWithLateBinding(repl.re, finalReplace, h.ReplacePlaceholders)
```

Comment:
```suggestion
					tr := replace.RegexpIndexFunc(re, func(src []byte, index []int) []byte {
						template := h.ReplacePlaceholders(finalReplace)
						return re.Expand(nil, []byte(template), src, index)
					})
```

No upstream changes are required. Just write you code here.

Reply:
Awesome! I fixed the typo (the `re` in your suggested change should be `repl.re`) and tested it. It works like a champ. Now the questions is "How about the conditional branch that deals with non-regexp string?"